packages/contract-utils/access-control/src/storage.rs

```diff
@@ -0,0 +1,519 @@
+    ensure_if_admin_or_admin_role(e, caller, role);
+
+    // Check if account has the role
+    if has_role(e, account, role).is_none() {
+        panic_with_error!(e, AccessControlError::AccountNotFound);
+    }
+
+    remove_from_role_enumeration(e, account, role);
+
+    let key = AccessControlStorageKey::HasRole(account.clone(), role.clone());
+    e.storage().persistent().remove(&key);
+
+    emit_role_revoked(e, role, account, caller);
+}
+
+/// Allows an account to renounce a role assigned to itself.
+/// Users can only renounce roles for their own account.
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+/// * `caller` - The address of the caller, must be the account that has the
+///   role.
+/// * `role` - The role to renounce.
+///
+/// # Errors
+///
+/// * `AccessControlError::AccountNotFound` - If the `caller` doesn't have the
+///   role.
+/// * refer to [`remove_from_role_enumeration()`] errors.
+///
+/// # Events
+///
+/// * topics - `["role_revoked", role: Symbol, account: Address]`
+/// * data - `[sender: Address]`
+///
+/// # Notes
+///
+/// * Authorization for `caller` is required.
+pub fn renounce_role(e: &Env, caller: &Address, role: &Symbol) {
+    caller.require_auth();
+    if has_role(e, caller, role).is_none() {
+        panic_with_error!(e, AccessControlError::AccountNotFound);
+    }
+
+    remove_from_role_enumeration(e, caller, role);
+
+    let key = AccessControlStorageKey::HasRole(caller.clone(), role.clone());
+    e.storage().persistent().remove(&key);
+
+    emit_role_revoked(e, role, caller, caller);
+}
+
+/// Initiates admin role transfer.
+/// Admin privileges for the current admin are not revoked until the
+/// recipient accepts the transfer.
+/// Overrides the previous pending transfer if there is one.
+///
+/// # Arguments
+///
+/// * `e` - Access to Soroban environment.
+/// * `admin` - The address of the admin.
+/// * `new_admin` - The account to transfer the admin privileges to.
+/// * `live_until_ledger` - The ledger number at which the pending transfer
+///   expires. If `live_until_ledger` is `0`, the pending transfer is cancelled.
+///   `live_until_ledger` argument is implicitly bounded by the maximum allowed
+///   TTL extension for a temporary storage entry and specifying a higher value
+///   will cause the code to panic.
+///
+/// # Errors
+///
+/// * `AccessControlError::Unauthorized` - If the `admin` is not actually the
+///   admin.
+/// * `AccessControlError::NoPendingAdminTransfer` - If tried to cancel the
+///   pending admin transfer when there is no pending admin transfer.
+///
+/// # Events
+///
+/// * topics - `["admin_transfer_started", current_admin: Address]`
+/// * data - `[new_admin: Address, live_until_ledger: u32]`
+///
+/// # Notes
+///
+/// * Authorization for `admin` is required.
+pub fn transfer_admin_role(e: &Env, admin: &Address, new_admin: &Address, live_until_ledger: u32) {
+    admin.require_auth();
+    if admin != &get_admin(e) {
```

Comment:
```suggestion
    if *admin != get_admin(e) {
```
It boils down to the same, but derefs feel more natural here imo.